[VendorPatches] Replace rtrim with substr method

Quazz opened the pull request:
Rtrim removes the characters rather than the word, resulting in edge cases such .xml.old being reduced to .xm 

This fix should be near equal performance wise, without rtrim drawbacks.

Should fix https://github.com/symplify/symplify/issues/3606

TomasVotruba commented:
Thanks :+1: 

Could you provide a test case, just to be sure it works?

Quazz commented:
> Thanks 👍
> 
> Could you provide a test case, just to be sure it works and someone won't revert in the future?

Never really worked with tests, hope it's okay to edit the existing one to look for 2 files to be returned now instead.

Also updated the original PR with the PHPStan feedback in mind and a slight improvement.

TomasVotruba commented:
Thanks. It seems this test would work before your change.

Could you add file name that was breaking in your case? That way we would replicate the same behaviour that caused the bug :+1: 

The CI needs to be green before merge, seems like PHPStan is failing

Quazz commented:
Fixed the typo. 

I added the files https://github.com/symplify/symplify/pull/3607/commits/6ee014be9ff1b748ae23e023f01b60e5dc99d661 
https://github.com/symplify/symplify/pull/3607/commits/d77cc83e872e403275f545ca1d4224efa172025f

TomasVotruba commented:
> I added the files 6ee014b
> d77cc83

I see. I had to check it fully - https://3v4l.org/aZ6t2

The test case makes sense :+1:  thank you
